### PR TITLE
Fix Algolia search indexing

### PR DESCRIPTION
**Pull request by CiaranMn**
The Algolia search indexing assumed that the `spec` would be in its folder, but it's been moved to a subfolder of `docs` in https://github.com/blockprotocol/blockprotocol/pull/365

This PR fixes the Algolia script to account for this.

The script needs more work, particularly to get better search result specificity and highlighting from Algolia - we may want to index individual _sections_ of documents rather than the whole thing. For a future PR.

**Note**: I have already re-indexed the production index to ensure the correct results are showing. This workflow only runs on `main`. To test it, run it locally, amending the index to `blockprotocol-testing`.

### CODE REVIEW

**Comment by kachkaev** [comment]
WDYT of moving this file from `scripts` to `site/scripts`? It seems to be related to that ‘app’ only. This can help us organise scripts in the future.

File: scripts/sync-algolia-index.ts [hunk 1]
```diff
@@ -27,33 +33,26 @@ const getFileInfos = (
...
+      return;
+    }
+
+    const inputPath = path.join(dirPath, "/", file);
+    const outputPath = `.\\output\\${path.join(
+      "./output/",
+      dirPath,
+      "/",
+      file,
+    )}`;
```
Comment (kachkaev) [inline scripts/sync-algolia-index.ts:53]: Hmmm not sure I understand why we have `path.join` inside ``` `` ``` 🤔

**Comment by CiaranMn** [comment]
> WDYT of moving this file from `scripts` to `site/scripts`? It seems to be related to that ‘app’ only. This can help us organise scripts in the future.

https://github.com/blockprotocol/blockprotocol/pull/375/commits/dbaa692c1d50cce63a0eeb63393bed803f8ba601

File: .github/workflows/algolia-upload-index.yml [hunk 2]
```diff
@@ -22,4 +22,4 @@ jobs:
       - run: yarn install
 
       - name: Sync Algolia Index
-        run: yarn exe scripts/sync-algolia-index.ts
+        run: yarn exe site/scripts/sync-algolia-index.ts
```
Comment (kachkaev) [inline .github/workflows/algolia-upload-index.yml:25]:
```suggestion
        run: yarn workspace @blockprotocol/site scripts/sync-algolia-index.ts
```

(both variants are generally equal, the main difference is `process.cwd()`

File: scripts/sync-algolia-index.ts (hunk 1 above)
Comment (CiaranMn) [reply scripts/sync-algolia-index.ts:53]: I expect we will heavily rewrite this soon as I'd rather not start improving bits like this

File: site/scripts/sync-algolia-index.ts [hunk 3]
```diff
@@ -19,6 +19,12 @@ type DocsFrontMatter = {
 
 type BlockProtocolFileTypes = "spec" | "docs";
```
Comment (kachkaev) [inline site/scripts/sync-algolia-index.ts:20]: Do we still need this?

Comment (CiaranMn) [reply site/scripts/sync-algolia-index.ts:20]: It is referred to in 3 places below

File: .github/workflows/algolia-upload-index.yml [hunk 4]
```diff
@@ -22,4 +22,4 @@ jobs:
       - run: yarn install
 
       - name: Sync Algolia Index
-        run: yarn exe scripts/sync-algolia-index.ts
+        run: yarn workspace @blockprotocol/site scripts/sync-algolia-index.ts
```
Comment (kachkaev) [inline .github/workflows/algolia-upload-index.yml:25]:
```suggestion
        run: yarn workspace @blockprotocol/site exe scripts/sync-algolia-index.ts
```

Oops, sorry I missed `exe` 😬😅

File: site/scripts/sync-algolia-index.ts (hunk 3 above)
Comment (kachkaev) [reply site/scripts/sync-algolia-index.ts:20]: I’m wondering if there still exists conceptual necessity in this type, given that `spec` is a sub-path of `/docs`

Comment (CiaranMn) [reply site/scripts/sync-algolia-index.ts:20]: Ah, right – the designs for results still distinguish between 'spec' results and 'docs' results (in breadcrumbs), so I think it's useful to keep them distinct, yes

**Comment by kachkaev** [comment]
@CiaranMn I’ve decided to merge this PR for you to fix ❌ on `main`. We can create a follow-up PR if anything is still missing!

**Pull request merged**